src/sorting/strand_sort.rs

```diff
@@ -0,0 +1,102 @@
+use std::collections::LinkedList;
+
+pub fn strand_sort(list: &LinkedList<i32>) -> LinkedList<i32> {
+    if list.is_empty() {
+        return LinkedList::new();
+    }
+
+    let mut ip = list.iter().cloned().collect::<LinkedList<i32>>();
+    let mut op = LinkedList::new();
+
+    while !ip.is_empty() {
+        let mut sublist = LinkedList::new();
+        sublist.push_back(ip.pop_front().unwrap());
+
+        let mut temp_list = LinkedList::new();
+        while let Some(val) = ip.pop_front() {
+            if val >= *sublist.back().unwrap() {
+                sublist.push_back(val);
+            } else {
+                temp_list.push_back(val);
+            }
+        }
+
+        let mut merged = LinkedList::new();
+        while let (Some(&op_val), Some(&sub_val)) = (op.front(), sublist.front()) {
+            if op_val <= sub_val {
+                merged.push_back(op.pop_front().unwrap());
+            } else {
+                merged.push_back(sublist.pop_front().unwrap());
+            }
+        }
+        merged.append(&mut op);
+        merged.append(&mut sublist);
```

Comment:
What do you think about creating two functions/procedures, something like `collect_strands` and `merge_strands`? This should improve the readability.